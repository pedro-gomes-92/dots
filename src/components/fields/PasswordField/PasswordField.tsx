import React, { FunctionComponent } from 'react';
import classnames from 'classnames';
import { BaseFieldTypeProps as BaseFieldProps, BaseField } from '../BaseField';
import { PasswordInput } from '../../inputs';

export interface PasswordFieldProps extends BaseFieldProps {}

export const PasswordField: FunctionComponent<PasswordFieldProps> = ({
  className,
  model,
  placeHolder,
  value,
  isReadOnly,
  onChange,
  size,
  ...rest
}: PasswordFieldProps): JSX.Element => (
  <BaseField className={classnames('is-password', className)} {...rest}>
    <PasswordInput
      placeHolder={placeHolder}
      model={model}
      value={value}
      onChange={onChange}
      isReadOnly={isReadOnly}
      size={size}
    />
  </BaseField>
);
