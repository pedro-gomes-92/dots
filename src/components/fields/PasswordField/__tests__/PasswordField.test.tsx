import React from 'react';
import { render } from '@testing-library/react';

import { PasswordField } from '../PasswordField';

describe('Given <PasswordField />', (): void => {
  describe('When initializes', (): void => {
    it('Then should match the snapshot', (): void => {
      const { container } = render(<PasswordField />);
      expect(container).toMatchSnapshot();
    });
  });
});
