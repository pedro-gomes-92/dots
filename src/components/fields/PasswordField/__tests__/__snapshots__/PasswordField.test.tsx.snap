// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given <PasswordField /> When initializes Then should match the snapshot 1`] = `
<div>
  <div
    class="field is-password"
  >
    <div
      class="field-body"
    >
      <div
        class="control"
      >
        <input
          class="input is-password"
          type="password"
        />
      </div>
    </div>
  </div>
</div>
`;
